docs/preview/counterExpression.md

```diff
@@ -0,0 +1,57 @@
+# Counter Expression
+
+## Overview
+
+This feature adds a new expression function, `counter`, which will atomically increment a variable value at the start of each build. The value of this variable will remain the same across phases, regardless of how or where those phases are executed.
+
+`counter` values are identified by their "prefix" which acts as a lookup key, meaning the same `counter` can be used in each phase or as a top-level variable, and it will be incremented exactly once for the build. The value of `counter` is unique to the build definition in which it is defined, regardless of the prefix.
+
+## Syntax
```

Comment:
It would be good to have small examples in this doc, that will make it easier to convert this document into user documentation. A small example showing how someone might use this functionality
```yaml
variables:
  version: counter('1.0.')
steps: ???
```